src/fetcher.ts

```diff
@@ -0,0 +1,42 @@
+import { Pair } from './entities/pair'
+import invariant from 'tiny-invariant'
+import ERC20 from './abis/hrc20_abi.json'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import JSBI from 'jsbi'
+import { getDefaultProvider } from '@ethersproject/providers'
+import { getNetwork } from '@ethersproject/networks'
+import { Contract } from '@ethersproject/contracts'
+import IUniswapV2Pair from '@uniswap/v2-core/build/IUniswapV2Pair.json'

```

Comment:
Last thing (maybe later): see if we can inline this into the current repo. I don't think this dependency is good.